src/components/DatePicker/index.ios.js

```diff
@@ -19,18 +20,29 @@ const datepickerPropTypes = {
 class Datepicker extends React.Component {
     constructor(props) {
         super(props);
-
+        const value = DateUtils.getDateAsText(props.value) || DateUtils.getDateAsText(props.defaultValue) || CONST.DATE.MOMENT_FORMAT_STRING;
         this.state = {
             isPickerVisible: false,
             selectedDate: props.value ? moment(props.value).toDate() : new Date(),
+            value,
         };
-
         this.showPicker = this.showPicker.bind(this);
         this.reset = this.reset.bind(this);
         this.selectDate = this.selectDate.bind(this);
         this.updateLocalDate = this.updateLocalDate.bind(this);
     }
 
+    componentDidUpdate() {
+        const dateValue = DateUtils.getDateAsText(this.props.value);
+        if (this.props.value === undefined || this.state.value === dateValue) {
+            return;
+        }
```

Comment:
Why is this logic different than the `index.android.js` one?

Reply:
Because the way we handle `value` state in android and ios are different. We are transforming the value to pass down to both the datepicker and textInput components